src/OpenBinary/reader.jl

```diff
@@ -355,30 +354,26 @@ function PSRI.open(
         data = data, # float cache
         relative_stage_skip = relative_stage_skip,
         io = io,
-        hs = hs,
-        skips = skips,
+        offset = header_size,
+        single_binary = single_binary,
     )
 
-    finalizer(ret) do x
-        if x.is_open
-            Base.close(x.io)
+    finalizer(ret) do ptr
+        if ptr.is_open
+            Base.close(ptr.io)
         end
-        x
+
+        return ptr
     end
```

Comment:
Maybe read more about it here https://docs.julialang.org/en/v1/base/base/#Base.finalizer. Does it need an `@async` call

Reply:
According to the docs, it needs. `Base.close` triggers a `flush` call, that is, an I/O operation. Thus, `@async` is required.
I added it.

Reply:
Let's leave like this and change in another PR, could you create an issue about it?